2022-11-04 console release (#286)
neondatabase/website: content/release-notes/2022-11-04.md

---
label: 'Console'
---

### What's new

> - UI: Frankfurt, Ohio, Singapore regions added.

Reviewer: ```suggestion
- UI: Added support for the US East (Ohio), Europe (Frankfurt), and Asia Pacific (Singapore) regions, allowing you to create Neon projects closer to your application servers. For more information about Neon's region support, see [Regions](https://neon.tech/docs/conceptual-guides/regions/).
```

> - UI: Branching support for the users with feature flag enabled.

Reviewer: ```suggestion
- UI: Updated Neon's branching capabilities. The following enhancements were introduced:
  - A **Branches** page was added to the Neon Console for creating and managing branches. 
  - Each project is now created with a root branch called `main`.
  - New branches are created as part of the project. You can create a branch from your project's root branch (`main`) or from another branch in the project. 
  - You can now define the data to include in a branch. You can include all data up to the current point in time or up to a past point in time.
  - Each branch now has a named endpoint, which is the compute instance associated with a branch. Connecting to a branch requires connecting to the branch's endpoint. You can obtain a connection string for a branch endpoint from the **Connection Details** widget on the Neon **Dashboard**.
  - For more information about Neon's branching capabilities, see [Branching](https://neon.tech/docs/conceptual-guides/branching/), and [Get started with branching](https://neon.tech/docs/get-started-with-neon/get-started-branching/).
  _** Note:**Neon Branching capabilities are not yet publicly available. If you would like to try this feature, contact us at  [iwantbranching@neon.tech](mailto:iwantbranching@neon.tech) describing your use case and requesting that Neon enable branching for your account._
```

> - UI: Support for postgres 15 keywords in query playground.

Reviewer: ```suggestion
- UI: Added support for [PostgreSQL 15 SQL Keywords](https://www.postgresql.org/docs/15/sql-keywords-appendix.html) to Neon's SQL Editor. Keywords are highlighted when entered into the SQL Editor.
```

> - UI: Topbar is shown in side drawer on smaller screens.

Reviewer: ```suggestion
- UI: Added support to display the horizontal navigation bar at the top of the Neon Console as a side drawer menu for small screens.
```

### Bug fixes

> - UI: Successful execution of multiqueries in query playground.

Reviewer: ```suggestion
- UI: Fixed an issue in the SQL Editor that caused an expected error not to be reported when rerunning a multi-statement query. 
```

> - UI: The .pgpass instructions on reset password popup have the port number.

Reviewer: ```suggestion
- UI: Updated the `.pgpass` configuration instructions provided after creating a project or resetting a password. The instructions did not include the required port number.
```

> - Control plane: users passing same oauth login challenge that have already been handled seconds time should now get redirect to final destination URL instead of 410 Gone error. 

Reviewer: ```suggestion
- Control Plane: Improved OAuth consent challenge handling. Submitting an OAuth consent challenge that was  handled previously now redirects a user to the final destination URL to complete the login or consent request instead of returning a `410 Gone` error.
```

> - Control plane: fixed memory leak on errors in background operations with context cancellation mechanism. 

Reviewer: ```suggestion
- Control plane: Implemented a fix for memory leaks that could occur for background operations started in a context that does not expire. Should such a background operation fail due to an error, resources used by the operation would not be cleaned up. 
```
